iocage/lib/ioc_create.py

```diff
@@ -100,126 +200,48 @@ def _create_jail(self, jail_uuid, location):
-                                     f"{self.release}"):
-                        iocage.lib.ioc_common.logit({
-                            "level"  : "EXCEPTION",
-                            "message": "You cannot clone a template, "
-                                       "use create -t instead."
-                        },
-                            _callback=self.callback,
-                            silent=self.silent)
-                    else:
-                        # Yep, self.release is actually the source jail.
-                        iocage.lib.ioc_common.logit({
-                            "level"  : "EXCEPTION",
-                            "message": f"Jail: {self.release} not found!"
-                        },
-                            _callback=self.callback,
-                            silent=self.silent)
-                else:
-                    iocage.lib.ioc_common.logit({
-                        "level"  : "EXCEPTION",
-                        "message": f"RELEASE: {self.release} not found!"
-                    },
-                        _callback=self.callback,
-                        silent=self.silent)
-
-            if not self.clone:
-                config = self.create_config(jail_uuid, cloned_release)
-            else:
-                clone_config = f"{self.iocroot}/jails/{jail_uuid}/config.json"
-                clone_fstab = f"{self.iocroot}/jails/{jail_uuid}/fstab"
+            _type = "releases"
+            config = self.config
+            basepath = f"{self.iocroot}/releases/{self.release}"
+            config_fn = self._get_release_config(jail_uuid, basepath)
 
         jail = f"{self.pool}/iocage/jails/{jail_uuid}/root"
+        self._su_check_call(jail_uuid, basepath, _type)
 
         if self.template:
-            try:
-                su.check_call(["zfs", "snapshot",
-                               f"{self.pool}/iocage/templates/{self.release}/"
-                               f"root@{jail_uuid}"], stderr=su.PIPE)
-            except su.CalledProcessError:
-                raise RuntimeError(f"Template: {self.release} not found!")
-
-            su.Popen(["zfs", "clone", "-p",
-                      f"{self.pool}/iocage/templates/{self.release}/root@"
-                      f"{jail_uuid}", jail], stdout=su.PIPE).communicate()
+            self._zfs_clone(f"{basepath}/root@{jail_uuid}", jail, "-p")
 
             # self.release is actually the templates name
-            config["release"] = iocage.lib.ioc_json.IOCJson(
-                f"{self.iocroot}/templates/{self.release}").json_get_value(
-                "release")
-            config["cloned_release"] = iocage.lib.ioc_json.IOCJson(
-                f"{self.iocroot}/templates/{self.release}").json_get_value(
-                "cloned_release")
+            config["release"] = config_fn("release")
+            config["cloned_release"] = config_fn("cloned_release")
+            config["type"] = "template"
```

Comment:
i reckon that if this "works" it means install packages below will fail
but i'd rather have that function shield itself from that by itself.